packages/common-all/src/service/ConfigService.ts

```diff
@@ -0,0 +1,245 @@
+import _ from "lodash";
+import { URI } from "vscode-uri";
+import { ConfigReadOpts, ConfigStore, IFileStore } from "../store";
+import { errAsync, okAsync } from "neverthrow";
+import { DendronConfig, DendronConfigValue, DVault } from "../types";
+import { ConfigUtils, DeepPartial } from "../utils";
+import { DendronError } from "../error";
+import * as YamlUtils from "../yaml";
+import { DConfigLegacy } from "../oneoff/ConfigCompat";
+
+export type ConfigServiceOpts = {
+  wsRoot: URI;
+  // the home directory.
+  // do not pass in home directory if the file store implementation has no concept of home directory.
+  // i.e. only pass it in for NodeJSFileStore
+  homeDir: URI | undefined;
+  fileStore: IFileStore;
+};
+
+export class ConfigService {
+  static _singleton: undefined | ConfigService;
+  public wsRoot;
+  public homeDir;
+  private _configStore: ConfigStore;
+  private _fileStore: IFileStore;
+
+  get configPath(): URI {
+    return this._configStore.configPath;
+  }
+
+  /** static */
+
+  static instance(opts?: ConfigServiceOpts) {
```

Comment:
Even if we want ConfigService has singleton like behavior, can we avoid exposing a static getter?